packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol

```diff
@@ -696,4 +708,220 @@ contract BridgeFacet is BaseConnextFacet {
+   * @return The amount left after the repayment
+   */
+  function _reconcileProcessPortal(
+    uint256 _amount,
+    address _local,
+    address _router,
+    bytes32 _transferId,
+    bytes32 _canonical
+  ) private returns (uint256) {
+    // When repaying a portal, should use available liquidity if there is not enough balance from
+    // the bridge. First, calculate the amount to be repaid in adopted asset then swap for exactly
+    // that amount. This prevents having to swap excess (i.e. from positive amm slippage) from debt
+    // repayment back into local asset to credit routers
+
+    // Calculates the amount to be repaid to the portal in adopted asset
+    (uint256 totalRepayAmount, uint256 backUnbackedAmount, uint256 portalFee) = _calculatePortalRepayment(
+      _amount + s.routerBalances[_router][_local],
+      s.aavePortalsTransfers[_transferId],
+      _transferId,
+      _local
+    );
+
+    // Swap for exact `totalRepayAmount` of adopted asset to repay aave
+    (uint256 amountIn, address adopted) = AssetLogic.swapFromLocalAssetIfNeededForExactOut(_local, totalRepayAmount);
+
+    // Edge case with some tokens: Example USDT in ETH Mainnet, after the backUnbacked call there could be a remaining allowance if not the whole amount is pulled by aave.
+    // Later, if we try to increase the allowance it will fail. USDT demands if allowance is not 0, it has to be set to 0 first.
+    // TODO: Should we call approve(0) and approve(totalRepayAmount) instead? or with a try catch to not affect gas on all cases?
+    // Example: https://github.com/aave/aave-v3-periphery/blob/ca184e5278bcbc10d28c3dbbc604041d7cfac50b/contracts/adapters/paraswap/ParaSwapRepayAdapter.sol#L138-L140
+    SafeERC20.safeIncreaseAllowance(IERC20(adopted), s.aavePool, totalRepayAmount);
+
+    (bool success, ) = s.aavePool.call(
+      abi.encodeWithSelector(IAavePool.backUnbacked.selector, adopted, backUnbackedAmount, portalFee)
+    );
+
+    if (success) {
+      emit AavePortalRepayment(_transferId, adopted, backUnbackedAmount, portalFee);
+    } else {
+      // Update the amount repaid to 0, so the amount is credited to the router
+      amountIn = 0;
+      emit AavePortalRepaymentDebt(_transferId, adopted, backUnbackedAmount, portalFee);
+    }
+
+    // TODO: Do we need to check balance before and after to get the exact amount paid and give the routers the rest?
+    // Aave accounts a global unbacked variable per asset for all, not by address/bridge.
+    // Someone can repay more than it should, so then a the moment of calling backUnbacked()
+    // aave can pull a smaller amount than backUnbackedAmount. So there will be an extra amount of assets that needs to be assigned
+    // See https://github.com/aave/aave-v3-core/blob/feb3f20885c73025f40cc272b59e7eacfaa02fe4/contracts/protocol/libraries/logic/BridgeLogic.sol#L121
+
+    // TODO: need to credit the router in the local asset, not the adopted asset
```

Comment:
Is this TODO a blocker to merge this PR?

Reply:
This is resolved -- this is what @liu-zhipeng was working on